x/marker/types/msg.go

```diff
@@ -577,3 +578,29 @@ func (msg *MsgSupplyIncreaseProposalRequest) GetSigners() []sdk.AccAddress {
 	addr := sdk.MustAccAddressFromBech32(msg.Authority)
 	return []sdk.AccAddress{addr}
 }
+
+// NewMsgUpdateRequiredAttributesRequest creates a MsgUpdateRequiredAttributesRequest
+func NewMsgUpdateRequiredAttributesRequest(denom string, transferAuthority sdk.AccAddress, removeRequiredAttributes, addRequiredAttributes []string) *MsgUpdateRequiredAttributesRequest {
+	return &MsgUpdateRequiredAttributesRequest{
+		Denom:                    denom,
+		TransferAuthority:        transferAuthority.String(),
+		RemoveRequiredAttributes: removeRequiredAttributes,
+		AddRequiredAttributes:    addRequiredAttributes,
+	}
+}
+
+func (msg MsgUpdateRequiredAttributesRequest) ValidateBasic() error {
+	if err := sdk.ValidateDenom(msg.Denom); err != nil {
+		return fmt.Errorf(err.Error())
+	}
+	if len(msg.AddRequiredAttributes) == 0 && len(msg.RemoveRequiredAttributes) == 0 {
+		return fmt.Errorf("both add and remove lists cannot be empty")
+	}
+	_, err := sdk.AccAddressFromBech32(msg.TransferAuthority)
+	return err
```

Comment:
Yeah, I thought my add/remove loops would fail if there were duplicates in the list.  Which still wouldn't account for duplicate entries of being in both lists.  Looks like the functions don't fail.  

It is easy enough to add to ValidateBasic to detect duplicates.